bot.py

```diff
@@ -0,0 +1,214 @@
+import os
+import re
+import sys
+from datetime import datetime
+import time
+
+import discord
+import dotenv
+import redis
+from discord.ext import commands
+from discord_slash import SlashCommand
+from discord_slash.utils.manage_commands import create_option, create_choice
+from dotenv import dotenv_values
+
+r = redis.Redis(host=dotenv_values("config.env")["HOSTNAME"], port=dotenv_values("config.env")["PORT"], db=dotenv_values("config.env")["DB"])
+
+now = datetime.now()  # current date and time
+date_time = now.strftime("%m/%d/%Y, %H:%M:%S")
+
+whitelisted = dotenv_values("config.env")["whitelisted"].split(",")
+bot = commands.Bot(command_prefix="redis ")
+slash = SlashCommand(bot, override_type=True, sync_commands=True)
+
+guild_ids = [int(dotenv_values("config.env")["guild_id"])]
+
+
+async def env_adj(key, value):
+    dotenv.set_key("config.env", str(key).upper(), value)
+
+
+@slash.slash(name="Reboot", description=f"", guild_ids=guild_ids)
+async def _reboot(ctx):
+    now = datetime.now()  # current date and time
+    result = discord.Embed(
+        title='Rebooting',
+        description=f'Execution time: {now}\n\n**ATTEMPTING REBOOT**',
+        colour=discord.Colour.teal()
+    )
+    result.set_footer(text="This is an open source project by @YonLiud at GitHub", icon_url="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png")
+    await ctx.send(embed=result)
+    time.sleep(2)
+    try:
+        if str(ctx.author.id) not in whitelisted:
+            raise Exception("01 Permission Denied! User id is not found in whitelisted")
+        output = "Verification Complete!\nRebooting..."
+    except Exception as e:
+        output = f"**An error has occurred**! Exception: ``{e}``"
+        if str(e).startswith("01 "):
+            output += "\n***🔴🔴This incident will be reported!🔴🔴***"
+        await ctx.send(output)
+    else:
+        await ctx.send(output)
+        try:
+            os.execv(sys.executable, ['python'] + sys.argv)
+        except Exception as e:
+            output = f"**An error has occurred**! Exception: ``{e}`` at {now}"
+            await ctx.message.delete()
+            await ctx.send(output)
+
+
+@slash.slash(name="config", description="Edit `config.env` file", guild_ids=guild_ids, options=[
+    create_option(
+        name="option",
+        description="Ed",
+        option_type=3,
+        required=True,
+
+        choices=[
+            create_choice(
+                name="Hostname",
+                value="HOSTNAME"
+            ),
+            create_choice(
+                name="Port",
+                value="PORT"
+            ),
+            create_choice(
+                name="Database",
+                value="DB"
+            )
+
+        ])
+])
```

Comment:
seems like it's just in the middle of the file, why not move it to its own module?

Reply:
part of the decorator, cannot shorten it